core/connectionlistmodel.cpp

```diff
@@ -21,7 +21,7 @@ int ConnectionListModel::rowCount(const QModelIndex&) const
 	return m_addresses.count();
 }
 
-void ConnectionListModel::addAddress(const QString address)
+void ConnectionListModel::addAddress(const QString &address)
```

Comment:
This change seems correct to me.

But be careful with slots - they might cross thread boundaries in which case you might end up with a dangling reference.

Also note that `QString`s have copy-on-write semantics, so copying is very cheap.  That forced COW is a bad idea anyway is a different matter...

Reply:
No, passing things to a slot in a different thread is safe: it will be copied when it is transferred between the even loops, so when you get the reference passed in it is from a QVariant.toString(), which will not change while you access it. The reference avoids one of this reference count up-/down cycles (which is otherwise cheap).

Reply:
I'm quite sure that I got crashes when doing the video thing, but I'll take your word for it...